crates/store/re_remote_store_types/proto/rerun/v0/remote_store.proto

```diff
@@ -27,13 +27,29 @@ message ObjectStorage {
 }
 
 message RegisterRecordingsResponse {
+    oneof response {
+        RegistrationSuccess success = 1;
+        RegistrationError error = 2;
+    }
```

Comment:
`RegisterRecordings` is a unary-unary endpoint, therefore I don't see the value in making it return a `oneof` response.

There are no streams involved, no RPC state to be kept alive, etc. We might as well let the client, as well as all the generic pieces of middleware in-between, know that the RPC flat out failed.

Imo:
* `RegisterRecordingsResponse` should only represent success (i.e. it's just `RegistrationSuccess`).
* `RegisterRecordings()` can fail, in which case it returns a gRPC/transport error where `details = RegistrationError{}`.

Reply:
that makes sense and it simplifies things on both sides